app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -1842,6 +1842,10 @@ export default class TopicController extends Controller.extend(
         }
         break;
       }
+      case "remove_allowed_user": {
+        this.router.transitionTo("userPrivateMessages", this.currentUser);
```

Comment:
I do wonder if we have to redirect the user away from the existing topic when the odds of a user being on the topic when the user is being removed is quite low. On top of that, I don't think it is a security concern for a user to be able to see the content of the topic that existed before the user was removed.

Reply:
I guess we could say that because it's unlikely we won't handle it. 😬  The reason I added it is the UI gets bricked. You can click stuff and nothing happens (because AJAX requests are failing without any feedback.)

Reply:
Then maybe we should just publish a simpler "request refresh" event instead so that the next click triggers a refresh of the page and the user is shown the not allowed to view topic message.

https://github.com/discourse/discourse/blob/e63a804ec93c23956978e2aec78b2710de1f1615/lib/discourse.rb#L826

Reply:
Added a system test. 🙏